src/proc.ts

```diff
@@ -71,10 +72,70 @@ export interface Subprocess {
 export interface BuildCommand {
   command: string;
   args?: string[];
-  build_env?: {[key: string]: string};
+  build_env?: EnvironmentVariables;
 }
 
-export interface EnvironmentVariables { [key: string]: string }
+export class EnvironmentVariables {
```

Comment:
Issue:  Need a way to keep "null"s because of the way presets use them to erase variables.  The way things are merged now, "null"s will be deleted immediately, but they need to stay around until we access the `.fullEnvironment` at which point they should be removed.  Could add a member variable to track whether the environment wants this behavior since most don't and it's a perf hit (though maybe insignificant) to have to scrub the environment before converting back to Dictionary<string>.

Issue: This class should be in its own module.

Reply:
How about implement EnvironmentVariables like NodeJS.ProcessEnv,
Getting EnvironmentVariables  as a alias of NodeJS.ProcessEnv,
as NodeJS.ProcessEnv handling windows case-insensitve properly